druntime/src/rt/aaA.d

```diff
@@ -974,3 +974,22 @@ unittest
     GC.runFinalizers((cast(char*)(&entryDtor))[0 .. 1]);
     assert(T.dtor == 6 && T.postblit == 2);
 }
+
+// Ensure the newaa struct layout (used for static initialization) is in sync
+unittest
+{
+    import newaa = core.internal.newaa;
+    static assert(newaa.Impl.sizeof == Impl.sizeof);
+    // ensure compatible types and offsets
+    static foreach (i; 0 .. Impl.tupleof.length)
+    {
+        // for bucket array and Flags, we "compatible" types, not exactly the same types.
```

Comment:
What counts as a compatible type.

Reply:
i.e. the same layout, different types. The Bucket struct should be identical in both implementations.

Note that this test isn't adequate to ensure that, as you could e.g. reorder the fields and this would still pass. But this is better than nothing.